src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,119 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int BASIC_CAPACITY = 10;
+    private static final double RE_SIZE = 1.5;
+    private static final int ZERO = 0;
+    private int size;
+
+    private T[] elementData = (T[]) new Object[BASIC_CAPACITY];
+
     @Override
     public void add(T value) {
-
+        if (size < elementData.length) {
+            elementData[size] = value;
+            size++;
+        } else {
+            elementData = Arrays.copyOf(elementData, newSize(elementData.length));
+            elementData[size] = value;
+            size++;
+        }
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < ZERO || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Index is out of ArrayList length");
+        }
 
+        if (size == elementData.length) {
+            elementData = Arrays.copyOf(elementData, newSize(elementData.length));
+        }
+
+        if (index == size) {
+            elementData[size] = value;
+        } else {
+            System.arraycopy(elementData, index, elementData, index + 1, size - index);
+            elementData[index] = value;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        if (list != null && !list.isEmpty()) {
+
+            if (elementData.length < size + list.size()) {
+                elementData = Arrays.copyOf(elementData, size + list.size());
+            }
 
+            for (int i = ZERO; i < list.size(); i++) {
+                T element = list.get(i);
+                elementData[size++] = element;
+            }
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (index >= ZERO && index < size) {
+            return elementData[index];
+        }
+        throw new ArrayListIndexOutOfBoundsException("Index is out of ArrayList length");
```

Comment:
include index and size in the message
extract repeated index check to a private method